src/System.Drawing.Primitives/tests/SizeTests.netcoreapp.cs

```diff
@@ -0,0 +1,132 @@
+        }
+
+
+        [Theory]
+        [InlineData(1000, 2400.933f)]
+        [InlineData(int.MaxValue, 0.0f)]
+        [InlineData(int.MaxValue, 1.0f)]
+        [InlineData(int.MaxValue, -1.0f)]
+        [InlineData(int.MaxValue, float.MaxValue)]
+        [InlineData(int.MinValue, 0.0f)]
+        [InlineData(int.MinValue, 1.0f)]
+        [InlineData(int.MinValue, -1.0f)]
+        [InlineData(int.MinValue, float.MinValue)]
+        [InlineData(int.MaxValue, float.MinValue)]
+        [InlineData(int.MinValue, float.MaxValue)]
+        public void MultiplicationTestSizeFloat(int value1, float value2)
+        {
+            Size sz1 = new Size(value1, value1);
+            SizeF mulExpected;
+
+            unchecked
+            {
+                mulExpected = new SizeF(value1 * value2, value1 * value2);
+            }
+
+            Assert.Equal(mulExpected, sz1 * value2);
+            Assert.Equal(mulExpected, value2 * sz1);
+        }
+
+        [Fact]
+        public void DivideByZeroChecks()
+        {
+            Size size = new Size(100, 100);
+            Assert.Throws<DivideByZeroException>(() => size / 0);
+
+            float invDiv = 1 / 0.0f;
+            SizeF expectedSizeF = new SizeF(float.PositiveInfinity, float.PositiveInfinity);
+            Assert.Equal(expectedSizeF, size * invDiv);
+        }
+
+        [Theory]
+        [InlineData(0, 1)]
+        [InlineData(1, 1)]
+        [InlineData(-1, 1)]
+        [InlineData(1, -1)]
+        [InlineData(-1, -1)]
+        [InlineData(int.MaxValue, int.MaxValue)]
+        [InlineData(int.MaxValue, int.MinValue)]
+        [InlineData(int.MinValue, int.MaxValue)]
+        [InlineData(int.MinValue, int.MinValue)]
+        [InlineData(int.MaxValue, 1)]
+        [InlineData(int.MinValue, 1)]
+        [InlineData(int.MaxValue, -1)]
+        public void DivideTestSizeInt(int value1, int value2)
+        {
+            Size size = new Size(value1, value1);
+            Size expected;
+
+            int invDiv = 1 / value2;
+            expected = new Size(value1 * invDiv, value1 * invDiv);
+
+            Assert.Equal(expected, size / value2);
+        }
+
+        [Theory]
+        [InlineData(0, 1.0f)]
+        [InlineData(1, 1.0f)]
+        [InlineData(-1, 1.0f)]
+        [InlineData(1, -1.0f)]
+        [InlineData(-1, -1.0f)]
+        [InlineData(int.MaxValue, float.MaxValue)]
+        [InlineData(int.MaxValue, float.MinValue)]
+        [InlineData(int.MinValue, float.MaxValue)]
+        [InlineData(int.MinValue, float.MinValue)]
+        [InlineData(int.MaxValue, 1.0f)]
+        [InlineData(int.MinValue, 1.0f)]
+        [InlineData(int.MaxValue, -1.0f)]
+        [InlineData(int.MinValue, -1.0f)]
+        public void DivideTestSizeFloat(int value1, float value2)
+        {
+            SizeF size = new SizeF(value1, value1);
+            SizeF expected;
+
+            float invDiv = 1.0f / value2;
+
+            expected = new SizeF(value1 * invDiv, value1 * invDiv);
```

Comment:
I think I tried asking that in [this](https://github.com/dotnet/corefx/pull/20052#discussion_r117809814) comment.

Between, the test case fails if I do not use reciprocal multiplication in the test case. I could not understand why we would want to use an approach outside (test case) which is different than inside the operator. Because the operator is just a syntactic sugar and supposed to behave on _whole_ in identical way as it would behave with _parts_; and really doesn't change the mathematics...?

For the following inputs.
```
        [InlineData(int.MaxValue, float.MaxValue)]
        [InlineData(int.MaxValue, float.MinValue)]
        [InlineData(int.MinValue, float.MaxValue)]
        [InlineData(int.MinValue, float.MinValue)]
```
The output is this,
```  D:\WinCPP\CoreFx\bin/AnyOS.AnyCPU.Debug/System.Drawing.Primitives.Tests/netcoreapp//RunTests.cmd D:\WinCPP\CoreFx\bin/testhost/netcoreapp-Windows_NT-Debug-x64/
  Using D:\WinCPP\CoreFx\bin\testhost\netcoreapp-Windows_NT-Debug-x64\ as the test runtime folder.
  Executing in D:\WinCPP\corefx\bin\AnyOS.AnyCPU.Debug\System.Drawing.Primitives.Tests\netcoreapp\
  Running tests... Start time:  1:18:29.82
  xUnit.net console test runner (64-bit .NET Core)
  Copyright (C) 2014 Outercurve Foundation.

  Discovering: System.Drawing.Primitives.Tests
  Discovered:  System.Drawing.Primitives.Tests
  Starting:    System.Drawing.Primitives.Tests
     System.Drawing.PrimitivesTests.SizeTests.DivideTestSizeFloat(value1: 2147483647, value2: 3.402823E+38) [FAIL]
        Assert.Equal() Failure
        Expected: {Width=6.310888E-30, Height=6.310888E-30}
        Actual:   {Width=6.310887E-30, Height=6.310887E-30}
        Stack Trace:
           D:\WinCPP\CoreFx\src\System.Drawing.Primitives\tests\SizeTests.netcoreapp.cs(128,0): at System.Drawing.PrimitivesTests.SizeTests.DivideTestSizeFloat(Int32 value1, Singl
  e value2)
     System.Drawing.PrimitivesTests.SizeTests.DivideTestSizeFloat(value1: 2147483647, value2: -3.402823E+38) [FAIL]
        Assert.Equal() Failure
        Expected: {Width=-6.310888E-30, Height=-6.310888E-30}
        Actual:   {Width=-6.310887E-30, Height=-6.310887E-30}
        Stack Trace:
           D:\WinCPP\CoreFx\src\System.Drawing.Primitives\tests\SizeTests.netcoreapp.cs(128,0): at System.Drawing.PrimitivesTests.SizeTests.DivideTestSizeFloat(Int32 value1, Singl
  e value2)
     System.Drawing.PrimitivesTests.SizeTests.DivideTestSizeFloat(value1: -2147483648, value2: 3.402823E+38) [FAIL]
        Assert.Equal() Failure
        Expected: {Width=-6.310888E-30, Height=-6.310888E-30}
        Actual:   {Width=-6.310887E-30, Height=-6.310887E-30}
        Stack Trace:
           D:\WinCPP\CoreFx\src\System.Drawing.Primitives\tests\SizeTests.netcoreapp.cs(128,0): at System.Drawing.PrimitivesTests.SizeTests.DivideTestSizeFloat(Int32 value1, Singl
  e value2)
     System.Drawing.PrimitivesTests.SizeTests.DivideTestSizeFloat(value1: -2147483648, value2: -3.402823E+38) [FAIL]
        Assert.Equal() Failure
        Expected: {Width=6.310888E-30, Height=6.310888E-30}
        Actual:   {Width=6.310887E-30, Height=6.310887E-30}
        Stack Trace:
           D:\WinCPP\CoreFx\src\System.Drawing.Primitives\tests\SizeTests.netcoreapp.cs(128,0): at System.Drawing.PrimitivesTests.SizeTests.DivideTestSizeFloat(Int32 value1, Singl
  e value2)
  Finished:    System.Drawing.Primitives.Tests
```